### ISSUE TITLE
RFE: fetch a unit file from a URL

### ISSUE THREAD

**Issue by omkensey** [systemd/systemd#20565]
**Is your feature request related to a problem? Please describe.**
When a large number of systems are using the same custom unit file, it would often be convenient to have them able to pull it from a URL (such as a file repository or artifact service) instead of needing to either embed it in the system image or push it out to every system.

There is precedent for systemd components using a remote service URL -- `systemd-journal-upload` sends journal events to a remote URL, and `systemd-journal-remote` receives events from one.

**Describe the solution you'd like**
A new `systemctl` subcommand (I'd suggest `systemctl fetch` but anything is fine) that requests a unit file from a URL and writes it to a relevant path in the filesystem.  Optionally (e.g. by using the flag `--now`) the system manager configuration is reloaded.  The response is verified as a valid unit file before writing (a result that is not a valid unit file should throw an error and make no changes).  Other options like which path in the filesystem to write the unit file to or whether to throw an error instead of overwriting an existing unit file could be controlled by other flags or implied from the request or its result (e.g. the specific filename could be given by a flag, or implied from the request URL).

**Describe alternatives you've considered**
It would be possible to script at least a basic version of this e.g. by using `curl` and existing `systemctl` commands, but that's not as portable as simply having the functionality built into `systemctl`.  In particular, not every system has any given command like `curl` or `wget` commonly used to fetch data over HTTP, and not every implementation of those commands is identical, so half of such a script would become a mess of looking for and testing various utilities and changing behavior depending on which one is finally found.

**The systemd version you checked that didn't have the feature you are asking for**
I searched the systemd/systemd GitHub repo (search string "unit url") and didn't see anything that looked like it implements this already.

**Comment by bluca** [comment]
In general the unit file without the corresponding executables will be pretty useless, no? For which we have portable services https://systemd.io/PORTABLE_SERVICES/ but maybe we are missing support in `machinectl pull` for these

**Comment by poettering** [comment]
Yeah, I agree with @bluca, a unit file alone doesn't really help does it? i.e. unit files and the binaries they decalre to be invoked should be shipped and delivered together, solving only one of the two doesn't really fix the problem?

And yes, we should add "portablectl pull" so that people can easily download portable service images in one step and place them at the right place.

**Comment by omkensey** [comment]
The particular case I'm thinking of would involve a binary pulled and run by something already on the system, e.g. Docker/podman -- so just pulling the unit file on demand would actually still be highly useful since it would contain instructions for handling pull/refresh of the app binary.
